test/e2e/operator_test.go

```diff
@@ -1291,6 +1300,79 @@ func TestInternalLoadBalancerGlobalAccessGCP(t *testing.T) {
 	}
 }
 
+// TestAWSResourceTagsChanged tests the functionality of updating AWS resource tags
+// in the infrastructure configuration and validates that the expected
+// awsLBAdditionalResourceTags is set correctly on the
+// loadBalancer service associated with the default Ingress Controller.
+//
+// This test is a serial test because it modifies the cluster infrastructure config and
+// therefore should not run in parallel with other tests.
+func TestAWSResourceTagsChanged(t *testing.T) {
+
+	if infraConfig.Status.Platform != "AWS" {
+		t.Skipf("test skipped on platform %q", infraConfig.Status.Platform)
+	}
+	if err := waitForIngressControllerCondition(t, kclient, 10*time.Second, defaultName, defaultAvailableConditions...); err != nil {
+		t.Errorf("did not get expected conditions: %v", err)
+	}
+
+	t.Log("Updating AWS ResourceTags in the cluster infrastructure config")
+	retryErr := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
Because we want to update the status of the Infrastructure config, instead of spec. I've moved the status update logic to a new function `updateInfrastructureConfigStatusWithRetryOnConflict` for better understanding.